pkg/admission/apibinding/apibinding_admission.go

```diff
@@ -0,0 +1,225 @@
+	}
+	apiExportClusterName := helper.EncodeOrganizationAndClusterWorkspace(org, apiBinding.Spec.Reference.Workspace.WorkspaceName)
+
+	// Access check
+	if err := o.checkAPIExportAccess(ctx, a.GetUserInfo(), apiExportClusterName, apiBinding.Spec.Reference.Workspace.ExportName); err != nil {
+		action := "create"
+		if a.GetOperation() == admission.Update {
+			action = "update"
+		}
+		return admission.NewForbidden(a, fmt.Errorf("unable to %s APIImport: %w", action, err))
+	}
+
+	return nil
+}
+
+func (o *apiBindingAdmission) checkAPIExportAccess(ctx context.Context, user user.Info, clusterName, exportName string) error {
+	authz, err := o.createAuthorizer(clusterName, o.kubeClusterClient)
+	if err != nil {
+		// Logging a more specific error for the operator
+		klog.Errorf("error creating authorizer from delegating authorizer config: %v", err)
+		// Returning a less specific error to the end user
+		return errors.New("unable to authorize request")
+	}
+
+	bindAttr := authorizer.AttributesRecord{
+		User:            user,
+		Verb:            "bind",
+		APIGroup:        apisv1alpha1.SchemeGroupVersion.Group,
+		APIVersion:      apisv1alpha1.SchemeGroupVersion.Version,
+		Resource:        "apiexports",
+		Name:            exportName,
+		ResourceRequest: true,
+	}
+
+	if decision, _, err := authz.Authorize(ctx, bindAttr); err != nil {
+		return fmt.Errorf("unable to determine access to apiexports: %w", err)
+	} else if decision != authorizer.DecisionAllow {
+		return errors.New("missing verb='bind' permission on apiexports")
+	}
+
+	return nil
+}
+
+// Admit applies the default APIBinding initializer to an APIBinding when it is transitioning to the
+// Initializing phase.
+func (o *apiBindingAdmission) Admit(_ context.Context, a admission.Attributes, _ admission.ObjectInterfaces) error {
+	if a.GetResource().GroupResource() != apisv1alpha1.Resource("apibindings") {
+		return nil
+	}
+
+	if a.GetOperation() != admission.Update {
+		return nil
+	}
+
+	u, ok := a.GetObject().(*unstructured.Unstructured)
+	if !ok {
+		return nil
+	}
+	obj, err := kcpadmissionhelpers.DecodeUnstructured(u)
+	if err != nil {
+		// nolint: nilerr
+		return nil // only work on unstructured APIBindings
+	}
+	apiBinding, ok := obj.(*apisv1alpha1.APIBinding)
+	if !ok {
+		// nolint: nilerr
+		return nil // only work on unstructured APIBindings
+	}
+
+	obj, err = kcpadmissionhelpers.NativeObject(a.GetOldObject())
+	if err != nil {
+		return fmt.Errorf("unexpected unknown old object, got %v, expected APIBinding", a.GetOldObject().GetObjectKind().GroupVersionKind().Kind)
+	}
+	old, ok := obj.(*apisv1alpha1.APIBinding)
+	if !ok {
+		return fmt.Errorf("unexpected unknown old object, got %v, expected APIBinding", obj.GetObjectKind().GroupVersionKind().Kind)
+	}
+
+	// we only admit at state transition to initializing
+	transitioningToInitializing :=
+		old.Status.Phase != apisv1alpha1.APIBindingPhaseBinding &&
+			apiBinding.Status.Phase == apisv1alpha1.APIBindingPhaseBinding
```

Comment:
do we have a test that we only move forward in the phases?

Reply:
I originally had that, but because you can rebind, I think the only transition it makes sense to forbid is going back to `""`. I have `Update: transition backwards binding to '' fails`. I can add `Update: transition backwards bound to '' fails` if you'd like?

Reply:
that's fine